pkg/hub/localizer/localizer_test.go

```diff
@@ -0,0 +1,412 @@
+		if string(toYaml) != string(tt.want) {
+			t.Errorf("%v: Got %s WANT %s", tt.testName, string(output), string(tt.want))
+		}
+	}
+}
+
+func TestPatcherWithJsonPatch(t *testing.T) {
+	deployment := []byte(`
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app: nginx
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - env:
+        - name: test
+          value: 123
+        image: nginx
+        name: nginx
+        ports:
+        - containerPort: 80
+`)
+
+	tests := []struct {
+		testName string
+		input    []byte
+		want     []byte
+	}{
+		{
+			testName: "Patch 3 patches",
+			input: []byte(`
+- path: "/spec/replicas"
+  value: 5
+  op: replace
+- path: "/spec/template/spec/containers/0/image"
+  value: "nginx:1.12.0-alpine"
+  op: replace
+- path: "/metadata/annotations"
+  op: "add"
+  value:
+    app: apple
+`),
+			want: []byte(`apiVersion: apps/v1
+kind: Deployment
+metadata:
+  annotations:
+    app: apple
+  name: nginx-deployment
+spec:
+  replicas: 5
+  selector:
+    matchLabels:
+      app: nginx
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - env:
+        - name: test
+          value: 123
+        image: nginx:1.12.0-alpine
+        name: nginx
+        ports:
+        - containerPort: 80
+`),
+		},
+		{
+			testName: "Json style patch",
+			input: []byte(`
+[
+  {
+    "path": "/spec/replicas",
+    "value": 5,
+    "op": "replace",
+  },
+  {
+    "path": "/spec/template/spec/containers/0/image",
+    "value": "nginx:1.12.0-alpine",
+    "op": "replace",
+  },
+  {
+    "path": "/metadata/annotations",
+    "op": "add",
+    "value": {
+      "app": "apple"
+    }
+  }
+]
+`),
+			want: []byte(`apiVersion: apps/v1
+kind: Deployment
+metadata:
+  annotations:
+    app: apple
+  name: nginx-deployment
+spec:
+  replicas: 5
+  selector:
+    matchLabels:
+      app: nginx
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - env:
+        - name: test
+          value: 123
+        image: nginx:1.12.0-alpine
+        name: nginx
+        ports:
+        - containerPort: 80
+`),
+		},
+	}
+	for _, tt := range tests {
+		override := appsapi.OverrideRule{
+			RuleName: tt.testName,
+			OverrideValue: map[string]string{
+				"patch": string(tt.input),
+			},
+		}
+		patcher, err := newPatcher(override, deployment)
+		if err != nil {
+			t.Errorf("%v: New Patcher error %v", tt.testName, err)
+		}
+
+		output, err := patcher.Apply()
+		if err != nil {
+			t.Errorf("%v: Apply Patcher error %v", tt.testName, err)
+		}
+
+		toYaml, _ := yaml.JSONToYAML(output)
+		if string(toYaml) != string(tt.want) {
+			t.Errorf("%v: Got %s WANT %s", tt.testName, string(output), string(tt.want))
+		}
+	}
+}
+
+func TestMergeChartValues(t *testing.T) {
+	src := `{"replicas":"3","image":"nginx:v0.6.0","a":{"b":"c"}}`
```

Comment:
better to keep `src`, `input`, `want` with the same syntax, `yaml` or `json`.

Reply:
ok, I will normalize all cases except one exception:  MergeChartValues can accept yaml style input ,